view.go

```diff
@@ -187,24 +201,32 @@ func (v *view) fetch(doneCh, successCh chan<- struct{}, errCh chan<- error) {
 		select {
 		case <-v.stopCh:
 			return
+		case <-v.ctx.Done():
+			return
 		default:
 		}
 
 		start := time.Now() // for rateLimiter below
 
 		if d, ok := v.dependency.(idep.QueryOptionsSetter); ok {
-			d.SetOptions(idep.QueryOptions{
+			opts := idep.QueryOptions{
 				AllowStale:   allowStale,
 				WaitTime:     v.blockWaitTime,
 				WaitIndex:    v.lastIndex,
 				DefaultLease: v.defaultLease,
-			})
+			}
+			opts = opts.SetContext(v.ctx)
+			d.SetOptions(opts)
 		}
 		data, rm, err := v.dependency.Fetch(v.clients)
 		if err != nil {
```

Comment:
I almost universally prefer switch statements over if/else blocks. 
Nothing technically wrong here, just a stylistic difference. 
No change required (unless, you know, you want to).

Reply:
Lol I actually had a switch statement here first when I thought I could compare `err == context.Canceled` to find out it doesn't work because it's nested errors.

For me, I usually use a switch when it's 3 or more cases based on equality since it's pretty clean in the amount of code, and prefer if-else for different true cases.